app/controllers/users_controller.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def create
```

Comment:
Thinking about the security of the app, I think we should also be protecting this endpoint. Unlike my comment about `user_exists`, the risk isn't about leaking data, but instead it's about allowing an unauthorized person write data about another user that they shouldn't be able to.

With the auth token that Auth0 sends us, we should validate that the user described by the token indeed has control of the email being passed in as the address here.

Reply:
I have made some updates here. Now this endpoint requires that the request include an Auth0 token, which is received when the f/e successfully auths against Auth0, and the user's designated auth0 user ID. Our rails app validates this token by hitting Auth0's servers; a success response includes the user ID for which the token has been assigned. the #create method compares this user ID against the one received from the frontend to ensure that they match.

One question is do we want to be storing the Auth0 userId in our database as well to handle cases where the user changes their email?

I think we may have discussed caching the `authorize` before_action, which I can work on doing in a future PR.

Reply:
> One question is do we want to be storing the Auth0 userId in our database as well to handle cases where the user changes their email?

Yes, I think we should store that, and that we should use it as the primary way to associate an Auth0 JWT to a User object in our DB. Thanks for bringing that up.

> Lastly, I think we may have discussed caching the `authorize` before_action, which I can work on doing in a future PR.

That sounds good to me, and it could even possibly be a ticket for a new volunteer, since I think it would be relatively narrowly scoped, and it's not on the critical path (even if the performance would be fairly bad without it).

Reply:
I went to create a caching ticket and then saw that it was already created, haha.  I've been on this project for too long (or not long enough?) https://github.com/orgs/ShelterTechSF/projects/10/views/1?pane=issue&itemId=42087115